chrome/content/coniks/context.js

```diff
@@ -0,0 +1,68 @@
+this.EXPORTED_SYMBOLS = ["AccountContext", "ConiksContext", "ConiksContexts"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("chrome://otr/content/helpers.js");
+
+var OS;
+
+class ConiksContext {
+  constructor(opts) {
+    opts = opts || {};
+
+    if (opts.account === null) {
+      return null;
+    }
+    this.account = opts.account;
+    // user policies
+    this.allowUnsignedKeyChange = opts.allowUnsignedKeyChange === null || opts.allowUnsignedKeyChange === undefined ?
+                                  true : opts.allowUnsignedKeyChange;
+    this.allowPublicLookup = opts.allowPublicLookup === undefined || opts.allowPublicLookup === undefined ?
+                             true : opts.allowPublicLookup;
+  }
+
+  setPolicy(ctx) {
+    this.allowUnsignedKeyChange = ctx.allowUnsignedKeyChange;
+    this.allowPublicLookup = ctx.allowPublicLookup;
+  }
+}
+
+class AccountContext extends ConiksContext {
+  constructor(opts) {
+    if (opts.account === null || opts.protocol === null) {
+      return null;
+    }
+    super(opts);
+    if (opts.fingerprint !== undefined || opts.fingerprint !== null) {
+      this.fingerprint = opts.fingerprint;
+    }
+    this.protocol = opts.protocol;
+    this.contacts = []; // array of ConiksContext
+    // this.TB = null;
+  }
+}
+
+var ConiksContexts = {
+  path: helpers.profilePath("coniks.contexts"),
+
+  loadFromFile: function(contexts) {
+    let data = helpers.readTextFile(ConiksContexts.path);
+    if (data === null || data === "") {
+      return;
+    }
+    let ctxs = JSON.parse(data);
+    ctxs.forEach(function(c) {
+      let acc = c.account + "@" + c.protocol;
+      if (contexts.has(acc)) {
+        let ctx = contexts.get(acc);
+        ctx.setPolicy(c);
+        ctx.fingerprint = c.fingerprint;
+      }
+    });
+  },
+
+  storeToFile: function(contexts) {
+    let data = JSON.stringify([...contexts.values()]);
+    helpers.writeTextFileAsync(data, ConiksContexts.path);
```

Comment:
Probably want to use the promisified methods for some of this stuff.
https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/OSFile.jsm/OS.File_for_the_main_thread